samza-core/src/main/java/org/apache/samza/table/batching/CompactBatch.java

```diff
@@ -68,8 +71,10 @@ public int size() {
 
     if (operation instanceof GetOperation) {
       queries.putIfAbsent(operation.getKey(), operation);
+    } else if (operation instanceof UpdateOperation) {
+      throw new BatchingNotSupportedException("Batching not supported for Updates with CompactBatches");
```

Comment:
Is it possible to throw the exception when we detect there is a .sendUpdateTo(table) and the config allows compact batching? Right now it's throwing in runtime which might be hard to debug.

Reply:
This exception is typically caught in the `updateAsync` method when batching fails and standard `updateAsync` method is called.
```
 try {
      return batchProcessor.processPutDeleteOrUpdateOperations(new UpdateOperation<>(key, update, args));
    } catch (BatchingNotSupportedException e) {
      return table.updateAsync(key, update, args);
    } catch (Exception e) {
      throw new SamzaException(e);
    }
```

Reply:
Oh, what I meant is that if we don't support update with certain table mode, like here batching with compaction, we should catch it during the pipeline building stage, instead of running stage. So I think we can add a check in SendUpdateToTableOperatorImpl or somewhere when input.sendUpdateToTable() is used (the high-level api). There we validate the compact batching is not turned on. That way the user will see the exception before running the pipeline.

Reply:
Right, that makes sense. It should be possible to do it within the constructor of `SendUpdateToTableOperatorImpl`. The only way to do that is to expose the `BatchProvider` from the `RemoteTable` and check if it is an instance of `CompactBatchProvider`.

Reply:
Hmm, how does user specify to use this compact batching? I think there should be either some api or configs to specify that. Could you double check?

Reply:
`RemoteTableDescriptor` has a `withBatchProvider` which can be used to specify a Complete/Compact batch right now.